scripts/stack_bounds.py

```diff
@@ -36,18 +42,30 @@
 # those which may not. Stack usage bounds are then given as conditional
 # expressions in terms of that argument.
 
-# To specify how to construct a non-recursive call graph, we give a
-# mapping from the functions that may recurse to the name of the
-# register used for the argument which controls recursion. We also give
-# lists of edges to include and exclude from the modified graph. An edge
-# is given as a pair of nodes. A node is given in one of two forms. For
-# a function which does not recurse, the node is simply given as the
-# name of the function. For each recursive function, we create two
+# To specify how to construct a non-recursive call graph, we first give
+# a mapping from each function that may recurse to a GraphLang
+# expression for the argument which controls recursion. The latter may
+# be a register or a stack slot. Each specified function is split into a
+# pair of nodes, with each node in the pair given as a tuple of the
+# function name and one of the two possible values of the boolean
+# controlling argument. In the resulting graph, each of the split nodes
+# initially has the same edges as in the original graph.
+
+# To make the graph non-recursive, we then give lists of edges to
+# include and exclude from the modified graph. An edge is given as a
```

Comment:
All this work makes me wonder if we shouldn't try to unroll the recursion in delete/finalise somehow. It's nice in the specs, but it's annoying in C and below. But a story for a different time.